### ISSUE TITLE
Version 1.3.1 breaks my application because of HEAD and OPTIONS

### ISSUE THREAD

**Issue by Musiksammler** [zendframework/zend-expressive-router#28]
Hello!

In my zend-expressive application I've routes like this:

    'routes' => [
        [
            'name' => 'company-add',
            'path' => '/company/add',
            'middleware' => Add::class,
            'allowed_methods' => ['GET'],
        ],
        [
            'name' => 'company-save',
            'path' => '/company/add',
            'middleware' => [
                Save::class,
                Add::class,
            ],
            'allowed_methods' => ['POST'],
        ],
    ]

As you can see I use the same path but with different methods. Before the update to 1.3.1 it worked fine.

But with 1.3.1 and the "always allow HEAD and OPTIONS" these configuration throws an exception

    Uncaught Zend\Expressive\Exception\DuplicateRouteException:
    Duplicate route detected; same name or path, and one or more HTTP methods intersect

I suspect that because HEAD and OPTIONS are always added to the allowed methods of a route the function \Zend\Expressive\Application::checkForDuplicateRoute(...) mis-checks a duplicate route for the same methods, HEAD and OPTIONS.

It might be useful to allow HEAD and OPTIONS in some cases but here it breaks an application. It might be better to enable HEAD and OPTIONS only when the config "allowed_methods" says so?

**Comment by acelaya** [comment]
Probably disabling the implicit usage of HEAD and OPTIONS in one of the routes would fix your problem (I'm having the same), but I don't see how to disable it via configuration, only on `Route` instances

**Comment by weierophinney** [comment]
I'll revert the changes introduced in 1.3.1, and issue a 1.3.2 release immediately.

The idea behind this was to ensure we always support HEAD and OPTIONS requests, as the specs say these should always work for a given resource. However, as I've discovered in working on the support for FastRoute, there are other ways to accomplish this.

I'll update this issue when I've got the 1.3.2 release completed.
